modules/git/repo_commit.go

```diff
@@ -202,36 +202,57 @@ func (repo *Repository) commitsByRange(id SHA1, page int) (*list.List, error) {
 }
 
 func (repo *Repository) searchCommits(id SHA1, opts SearchCommitsOptions) (*list.List, error) {
-	cmd := NewCommand("log", id.String(), "-100", "-i", prettyLogFormat)
-	if len(opts.Keywords) > 0 {
-		for _, v := range opts.Keywords {
-			cmd.AddArguments("--grep=" + v)
-		}
-	}
+	cmd := NewCommand("log", id.String(), "-100", prettyLogFormat)
+	args := []string{"-i"}
 	if len(opts.Authors) > 0 {
 		for _, v := range opts.Authors {
-			cmd.AddArguments("--author=" + v)
+			args = append(args, "--author="+v)
 		}
 	}
 	if len(opts.Committers) > 0 {
 		for _, v := range opts.Committers {
-			cmd.AddArguments("--committer=" + v)
+			args = append(args, "--committer="+v)
 		}
 	}
 	if len(opts.After) > 0 {
-		cmd.AddArguments("--after=" + opts.After)
+		args = append(args, "--after="+opts.After)
 	}
 	if len(opts.Before) > 0 {
-		cmd.AddArguments("--before=" + opts.Before)
+		args = append(args, "--before="+opts.Before)
 	}
 	if opts.All {
-		cmd.AddArguments("--all")
+		args = append(args, "--all")
+	}
+	if len(opts.Keywords) > 0 {
+		for _, v := range opts.Keywords {
+			cmd.AddArguments("--grep=" + v)
+		}
 	}
+	cmd.AddArguments(args...)
 	stdout, err := cmd.RunInDirBytes(repo.Path)
 	if err != nil {
 		return nil, err
 	}
-	return repo.parsePrettyFormatLogToList(stdout)
+	if len(stdout) != 0 {
+		stdout = append(stdout, '\n')
```

Comment:
If I'm getting this code right, you always add the first `'\n'` and then always remove the last one with`TrimSuffix()`. Perhaps it would be easier to prepend `'\n'` when required? e.g. change:
`stdout = append(stdout, hashMatching...)`
into
`stdout = append(stdout, '\n', hashMatching...)`
and remove the other appends & TrimSuffix?

Reply:
The issue with prepending is that `repo.parsePrettyFormatLogToList` is a very picky function. Having any empty line will result in a 500 error. By prepending, if the first command did not return any results, you could end up with an empty line at the start. You could check for if the line is empty beforehand but then you have an extra if statement that runs very time during the loop.

The important part is that the first `'\n'` is only added if the original command did not return anything.

Also, the multiple appends were needed because Go does not let you use `...` as a part, not the whole of a variadic function parameter.

Reply:
OK, I see it now.